en/docs/integration-guides/usecases/datamapper/read-csv-file-and-transform-to-xml-file.md

```diff
@@ -0,0 +1,601 @@
+
+      The tool shows a brief **“Creating…”** indicator, and then a new canvas opens.
+
+
+  4. Observe the Automation Canvas
+
+      Once created, the visual editor displays a blank automation diagram with a starting node.  
+      This represents the root of your pipeline where you will define the steps to read, map, and write data.
+
+
+## Step 6: Add a file read function to the automation flow
+
+In this step, you’ll configure the automation’s first functional node — a **Call Function** block — to read CSV data from the configured input path (`inputCSV`) into an array of CSV records (`CSV[]`).  
+This forms the initial data acquisition step of the workflow.
+
+
+  1. Open the Automation Flow
+
+      With the automation already created, you’ll now see a **Start** node connected to an **Error Handler** node.  
+      This is the default template for new automations.
+
+
+  2. Add the First Node
+
+    1. Hover your mouse below the **Start** node until a small **“+” button** appears.  
+    2. Click **+** to add a new operation.  
+        The system briefly displays `Generating next suggestion...` — this means it’s fetching available node types.
+
+
+  3. Choose the Node Type
+
+      Once suggestions appear, look to the **right-hand Node Panel** under the **Statement** category.  
+      You’ll see options such as:
+
+      - Declare Variable  
+      - Assign  
+      - **Call Function**  
+      - Map Data  
+
+Click **Call Function** to insert this node into the automation flow.
+
+
+## Step 7: Use `fileReadCsv` to load the csv file into memory
+
+Now that your Call Function flow is open, the next step is to configure a CSV File Read Function using the built-in `io:fileReadCsv()` operation.  
+This function reads CSV data from the path defined in your project’s configurables and loads it as a structured array of `CSV` records that can be mapped later.
+
+
+  1. Open the Function Search Panel
+
+    In the right-side **Functions** panel, scroll or search for a CSV-related function:
+    1. Click inside the **Search library functions** field.  
+    2. Type **csv** to filter the standard library.  
+    3. You’ll see a list under **data.csv** and **io** modules containing functions like:
+      - `parseStream`
+      - `parseString`
+      - `fileReadCsv`
+      - `fileWriteCsv`
+
+
+  2. Select `io:fileReadCsv`
+
+    Click **fileReadCsv** from the results list.  
+    A configuration panel opens on the right describing the function:
+
+    > *“Read file content as a CSV. When the expected data type is record[],  
+    > the first entry of the CSV file should contain matching headers.”*
+
+
+  3. Configure Parameters
+
+    You can see the fields as follows:
+
+    | Field | Description | Example |
+    |-------|--------------|----------|
+    | **Path** | Path to your CSV file | `${inputCSV}` (the variable created in Configurables) |
```

Comment:
Better to explain the low code way of gettint this. i.e.

<img width="738" height="552" alt="image" src="https://github.com/user-attachments/assets/8a3110f3-c2b1-49c7-8a4d-7ddab53782c0" />

Reply:
Btw, whether we select `Text` or `Expression` this steps show the error below. Check below. it seems the reason is https://github.com/wso2/docs-bi/pull/174/files#r2528267549.  This problem will go away if we remove the table as suggested in https://github.com/wso2/docs-bi/pull/174/files#r2528247287

<img width="674" height="587" alt="image" src="https://github.com/user-attachments/assets/c09c8116-7e15-4cd1-b76b-e5d434c63bc3" />